src/Database/Database.php

```diff
@@ -1537,11 +1537,15 @@ public function createRelationship(
                 throw new DuplicateException('Attribute already exists');
             }
 
-            if ($attribute->getAttribute('type') === self::VAR_RELATIONSHIP
-                && \strtolower($attribute->getAttribute('options')['twoWayKey']) === \strtolower($twoWayKey)
-                && $attribute->getAttribute('options')['relatedCollection'] === $relatedCollection->getId()
-            ) {
-                throw new DuplicateException('Related attribute already exists');
+            try {
+                if ($attribute->getAttribute('type') === self::VAR_RELATIONSHIP
+                    && \strtolower($attribute->getAttribute('options')['twoWayKey']) === \strtolower($twoWayKey)
+                    && $attribute->getAttribute('options')['relatedCollection'] === $relatedCollection->getId()
+                ) {
+                    throw new DuplicateException('Related attribute already exists');
+                }
+            } catch (DuplicateException $e) {
+                $twoWayKey ??= $collection->getId() . '_' . uniqid();
             }
```

Comment:
I don't think we need a try/catch here, we can use use the if:

```suggestion
            if ($attribute->getAttribute('type') === self::VAR_RELATIONSHIP
                && \strtolower($attribute->getAttribute('options')['twoWayKey']) === \strtolower($twoWayKey)
                && $attribute->getAttribute('options')['relatedCollection'] === $relatedCollection->getId()
            ) {
                $twoWayKey ??= $collection->getId() . '_' . uniqid();
            }
```